### PR TITLE
LOOP-3787 Warnings based on glucose input

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-3787

### CODE REVIEW

File: Loop/View Models/SimpleBolusViewModel.swift [hunk 1]
```diff
@@ -99,6 +104,22 @@ class SimpleBolusViewModel: ObservableObject {
...
         }
     }
+    
+    private func updateNotice() {
+        let minRecommendationGlucose = displayMealEntry ? LoopConstants.simpleBolusCalculatorMinGlucoseMealBolusRecommendation : LoopConstants.simpleBolusCalculatorMinGlucoseBolusRecommendation
+        
+        switch manualGlucoseQuantity {
+        case let g? where g < minRecommendationGlucose:
+            activeNotice = .glucoseBelowRecommendationLimit
+        case let g? where g < LoopConstants.simpleBolusCalculatorGlucoseWarningLimit:
```
Comment (darinkrauss) [inline Loop/View Models/SimpleBolusViewModel.swift:114]: [nit] This case is never going to happen with the existing constants. Is this intended?

**Review by darinkrauss — APPROVED** [review]
LGTM!

File: Loop/View Models/SimpleBolusViewModel.swift [hunk 2]
```diff
@@ -65,16 +67,19 @@ class SimpleBolusViewModel: ObservableObject {
     
     @Published var activeInsulin: String?
 
-    @Published var enteredCarbAmount: String = "" {
+    @Published var enteredCarbString: String = "" {
```
Comment (rickpasetto) [inline Loop/View Models/SimpleBolusViewModel.swift:70]: Nice 👍

File: Loop/View Models/SimpleBolusViewModel.swift [hunk 3]
```diff
@@ -99,6 +104,22 @@ class SimpleBolusViewModel: ObservableObject {
             _manualGlucoseString = newValue
         }
     }
+    
+    private func updateNotice() {
+        let minRecommendationGlucose = displayMealEntry ? LoopConstants.simpleBolusCalculatorMinGlucoseMealBolusRecommendation : LoopConstants.simpleBolusCalculatorMinGlucoseBolusRecommendation
```
Comment (rickpasetto) [inline Loop/View Models/SimpleBolusViewModel.swift:109]: [nit] are there unit tests that cover this new code?  If so, then 👍

File: Loop/View Models/SimpleBolusViewModel.swift [hunk 4]
```diff
@@ -142,6 +159,13 @@ class SimpleBolusViewModel: ObservableObject {
 
     private var bolus: HKQuantity? = nil
     
+    var bolusRecommended: Bool {
```
Comment (rickpasetto) [inline Loop/View Models/SimpleBolusViewModel.swift:162]: [nit] are there unit tests that cover this new code?  If so, then 👍

**Review by rickpasetto — APPROVED** [review]
LGTM! Only just a request to make sure there are unit tests covering new code.

File: Loop/View Models/SimpleBolusViewModel.swift (hunk 1 above)
Comment (ps2) [reply Loop/View Models/SimpleBolusViewModel.swift:114]: During meal bolus, `minRecommendationGlucose` will be 55, and `simpleBolusCalculatorGlucoseWarningLimit` is 70, so this case will be used if the entered glucose is between those.

File: Loop/View Models/SimpleBolusViewModel.swift (hunk 3 above)
Comment (ps2) [reply Loop/View Models/SimpleBolusViewModel.swift:109]: There are more unit tests coming in the next PR (which also has more alarms).

File: Loop/View Models/SimpleBolusViewModel.swift (hunk 4 above)
Comment (ps2) [reply Loop/View Models/SimpleBolusViewModel.swift:162]: I'll add a test for this in the next PR, since I've already made other changes here.

**Pull request merged**